From 9215871b246269cb7eee90b7a909c72e500b3745 Mon Sep 17 00:00:00 2001
From: cergo666 <cergo666@ya.ru>
Date: Tue, 16 Jun 2026 17:03:29 +0300
Subject: [PATCH] Florida: string_frida_rpc patch fix

---
 lib/base/rpc.vala | 30 ++++++++++++++++++++----------
 1 file changed, 20 insertions(+), 10 deletions(-)

diff --git a/lib/base/rpc.vala b/lib/base/rpc.vala
index 97edecf3..8c294cfc 100644
--- a/lib/base/rpc.vala
+++ b/lib/base/rpc.vala
@@ -22,6 +22,16 @@ namespace Frida {
 			Object (peer: peer);
 		}
 
+		public string getRpcStr (bool quote) {
+			string result = (string) GLib.Base64.decode (
+				(string) GLib.Base64.decode ("Wm5KcFpHRTZjbkJq"));
+
+			if (quote)
+				return "\"" + result + "\"";
+			else
+				return result;
+		}
+
 		/**
 		 * Calls a remote method and waits for its result.
 		 *
@@ -35,17 +45,17 @@ namespace Frida {
 
 			var request = new Json.Builder ();
 			request
-				.begin_array ()
-				.add_string_value ("frida:rpc")
-				.add_string_value (request_id)
-				.add_string_value ("call")
-				.add_string_value (method)
-				.begin_array ();
+			.begin_array ()
+			.add_string_value (getRpcStr (false))
+			.add_string_value (request_id)
+			.add_string_value ("call")
+			.add_string_value (method)
+			.begin_array ();
 			foreach (var arg in args)
 				request.add_value (arg);
 			request
-				.end_array ()
-				.end_array ();
+			.end_array ()
+			.end_array ();
 			string raw_request = Json.to_string (request.get_root (), false);
 
 			bool waiting = false;
@@ -96,7 +106,7 @@ namespace Frida {
 		 * @return true if the message was an RPC message and was handled
 		 */
 		public bool try_handle_message (string json) {
-			if (json.index_of ("\"frida:rpc\"") == -1)
+			if (json.index_of (getRpcStr (true)) == -1)
 				return false;
 
 			var parser = new Json.Parser ();
@@ -125,7 +135,7 @@ namespace Frida {
 				return false;
 
 			string? type = rpc_message.get_element (0).get_string ();
-			if (type == null || type != "frida:rpc")
+			if (type == null || type != getRpcStr (false))
 				return false;
 
 			var request_id_value = rpc_message.get_element (1);
-- 
2.54.0
